File: 이진영/BinarySearch/입국심사.js
// 특정 시간 안에 처리할 수 있는 사람 수
const getPerson = (times, time) => {
    let ret = 0;
    for (let t of times) {
        ret += Math.floor(time / t);
    }
    return ret;
}

function solution(n, times) {
    let low = 1;
    let high = times[times.length - 1] * n; // 중요! 왜 이렇게 설정해야 답이 나오는지 모르겠음...
    
    while (low <= high) {
        let mid = Math.floor((low + high) / 2);
        let person = getPerson(times, mid);
        
        if (person < n) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    
    return low;
}

File: 정민선/BinarySearch/입국심사.py
from collections import deque

def solution(n, times):
    answer = 0
    left = 1 # 문제가 1분 이상이니까 min 안쓰고 그냥 1
    right = max(times) * n
    person = 0
    
    while left <= right:
        person = 0
        mid = (left+right) // 2
        for time in times:
            person += mid // time
            if person >= n:
                break
        if person >= n: # 심사가 아직 안 끝난 경우 포함, 양쪽 끝 범위 점점 줄이거나 늘리기
            answer = mid
            right = mid-1
        elif person < n:
            left = mid+1
        
    return answer